Upcxx 2020.3 update

bonachea opened the pull request:
This is a preliminary PR for internal review

tgamblin committed:
bugfix: `spack.package.possible_dependencies()` should return virtuals

`spack.package.possible_dependencies()` was forgetting to pass the
`visited` dict to recursive calls; this fixes it

tgamblin committed:
refactor: `spack.package.possible_dependencies()` handles virtuals

- [x] move some logic for handling virtual packages from the `spack
  dependencies` command into `spack.package.possible_dependencies()`

- [x] rework possible dependencies tests so that expected and actual
  output are on the left/right respectively

michaelkuhn committed:
modules: store configure args during build (#11084)

This change stores packages' configure arguments during build and makes
use of them while refreshing module files. This fixes problems such as in
#10716.

sethrj committed:
Uniquify suffixes added to module names (#14920)

tldahlgren committed:
bugfix:  Add dependents when initializing spec from yaml (#15220)

The new build process, introduced in #13100 , relies on a spec's dependents in addition to their dependencies. Loading a spec from a yaml file was not initializing the dependents.

- [x] populate dependents when loading from yaml

tldahlgren committed:
bugfix: ensure proper dependency handling for package-only installs (#15197)

The distributed build PR (#13100) -- did not check the install status of dependencies when using the `--only package` option so would refuse to install a package with the claim that it had uninstalled dependencies whether that was the case or not.

- [x] add install status checks for the `--only package` case.
- [x] add initial set of tests

adamjstewart committed:
suite-sparse: fix installation for v5.X (#15326)

fixes #15184

GraphBLAS depends on m4 according to CMake error message
Do not use INSTALL= when compiling the library

gartung committed:
Add function replace_prefix_nullterm for use on mach-o rpaths. (#15347)

This recovers the old behavior of replace_prefix_bin that was
modified to work with elf binaries by prefixing os.sep to new prefix
until length is the same as old prefix.

alalazo committed:
Bugfix: resolve StopIteration message attribute failure (#15341)

Testing the install StopIteration exception resulted in an attribute error:

AttributeError: 'StopIteration' object has no attribute 'message'

This PR adds a unit test and resolves that error.

alalazo committed:
Correct pytest.raises matches to match (#15346)

tldahlgren committed:
Recover coverage from subprocesses during unit tests (#15354)

* Recover coverage from subprocesses during unit tests

aweits committed:
Fix for being able to 'spack load' packages that have been renamed. (#14348)

* Fix for being able to 'spack load' packages that have been renamed.

* tests: add test for 'spack load' of a installed, but renamed/deleted package

alalazo committed:
ArchSpec: fix semantics of satisfies when not concrete and strict is true (#15319)

tgamblin committed:
bugfix: installer.py shouldn't be executable (#15386)

This is a minor permission fix on the new installer.py introduced in #13100.

becker33 committed:
bugfix: fix install_missing_compilers option bug from v0.14.0 (#15416)

* bugfix: ensure bootstrapped compilers built before packages using the compiler

becker33 committed:
fetch_strategy: remove vestigial code (#15431)

tldahlgren committed:
bugfix: resolve undefined source_pkg_dir failure (#15339)

alalazo committed:
Creating versions from urls doesn't modify class attributes (#15452)

fixes #15449

Before this PR a call to pkg.url_for_version was modifying
class attributes determining different results for subsequents
calls and an error when the urls was empty.

tldahlgren committed:
testing:  increase installer coverage (#15237)

bonachea committed:
upcxx: Correct Cray ALPS detection

The executable is `aprun` not `alps`

eschnett committed:
gcc: Don't apply macOS 9.2.0 patch for 9.3.0 (#15493)

The patch neither applies nor is necessary.

bonachea committed:
upcxx: Add new version 2020.3.0

bonachea committed:
upcxx: fix install of 2020.3.0 on Cray

Unlike previous versions, 2020.3.0 needs (at build time), the full,
real PATH to the real C and C++ compilers that will remain valid
post-installation outside a Spack environment. The Spack compiler
wrappers on Cray are insufficient for this task because they break
post-install outside a Spack environment, ie:

```
Spack compiler must be run from Spack! Input 'SPACK_ENV_PATH' is missing.
```

Previous versions of UPC++ contained a bug (issue #287) on Cray causing the
provided compiler to be ignored in favor of a hard-coded 'cc' and 'CC' which
was carried through the installer un-expanded.  However this behavior was
deemed fragile, because it ignored valid configure-time overrides and was
incorrectly sensitive to the post-install user $PATH.  This defect was fixed in
2020.3.0 to correctly honor the compilers requested at configure time, but also
always force PATH expansion of them at configure time.

This defect happenned to hide the fact Spack was changing the compiler identity
between install and use (by changing $PATH from one containing Spack's wrappers
to a normal use-time PATH lacking them). However with the bug fixed the Spack
wrappers no longer generate a usable install.

mdorier committed:
unqlite: added package at v1.1.9 (#15501)

eschnett committed:
cmake: added  v3.16.5 (#15492)

eschnett committed:
ncurses: added  v6.2 (#15494)

eschnett committed:
swig: added v4.0.1 (#15496)

melven committed:
tixi: added package at v3.0.3 (#15490)

gartung committed:
Buildcache: Install into non-default directory layouts (#13797)

* Buildcache: Install into non-default directory layouts

Store a dictionary mapping of original dependency prefixes to dependency hashes

Use the loaded spec to grab the new dependency prefixes in the new directory layout.

Map the original dependency prefixes to the new dependency prefixes using the dependency hashes.

Use the dependency prefixes map to replace original rpaths with new rpaths preserving the order.
For mach-o binaries, use the dependency prefixes map to replace the dependency library entires for libraries and executables and the replace the library id for libraries.

On Linux, patchelf is used to replace the rpaths of elf binaries.
On macOS, install_name_tool is used to replace the rpaths and  dependency libraries  of mach-o binaries and the id of mach-o libraries.
On Linux, macholib is used to replace the dependency libraries of mach-o binaries and the id of mach-o libraries.

Binary text with padding replacement is attempted for all binaries for the following paths:
spack layout root
spack prefix
sbang script location
dependency prefixes
package prefix
 Text replacement is attempted for all text files using the paths above.

Symbolic links to the absolute path of the package install prefix are replaced, all others produce warnings.

eschnett committed:
silo: depends on hdf5@:1.10.999 (#15495)

Silo does not build with HDF5 1.12.

sethrj committed:
CMakePackage: convert variants to CMake arguments (#14376)

Add a 'define_from_variant` helper function to CMake-based Spack
packages to convert package variants into CMake arguments. For
example:

  args.append('-DFOO=%s' % ('ON' if '+foo' in self.spec else 'OFF'))

can be replaced with:

  args.append(self.define_from_variant('foo'))

The following conversions are handled automatically:

* Flag variants will be converted to CMake booleans
* Multivalued variants will be converted to semicolon-separated strings
* Other variant values are converted to CMake string arguments

This also adds a 'define' helper method to convert any variable to
a CMake argument. It has the same conversion rules as
'define_from_variant' (but operates directly on values rather than
requiring the user to supply the name of a package variant).

gartung committed:
When replacing path text in ELF binaries do not include null byte padding in search string (#15508)

sethrj committed:
veccore: added package at v0.6.0 (#15472)

gartung committed:
Warn only if link target is not relative and outside of the install prefix (#15512)

bonachea committed:
upcxx: Add a run_tests attribute for post-install test_install

eugeneswalker committed:
cap binutils version @:2.33.1 due to build failures with 2.34 (#15504)

bonachea committed:
upcxx: Add an mpi variant

This defaults to off because mpi-conduit is almost always the wrong
choice for performance, and MPI spawner currently requires additional
configuration to work smoothly with UPC++.

alalazo committed:
Module files won't use CPATH by default, but language specific vars (#14749)

fixes #11555

Every path in CPATH is equivalent to a -I path to the compiler,
while every path in *_INCLUDE_PATH is equivalent to -isystem.
The latter avoids the noise due to warnings coming from 3rd party
libraries that a project depends on.

Added INCLUDE env variable (Intel Fortran, .mod files)

darmac committed:
nfs-utils: added package at v2.4.2 (#15491)

sethrj committed:
Update and exetend VecGeom (#14520)

* Add new vecgeom versions, add cuda support, automate target options

* Add ROOT, GDML, and external VecCore support to VecGeom

* Address reviewer comments

* Update vecgeom for CUDA

* Update versions

t-karatsu committed:
geant4: Removed unnecessary definition of xerces-c. (#15513)

wspear committed:
Corrected compiler filtering for TAU makefiles (#15342)

* Implemented working file filtering to replace spack compiler wrapper with real compiler.

* Using string=True instead of re.escape. Using self.prefix.lib instead of appending /lib.

Co-authored-by: Wyatt Spear <wspear@cs.uoregon.edu>

gartung committed:
Remove PatchelfError and change patchelf error message to a warning. (#15518)

skosukhin committed:
MPICH: optional libxml2 support; NAG patches (#15235)

* Add patches when building with NAG
* Make libxml2 support optional. Also include conflict for
  @:3.2~hydra+libxml2 since @:3.2~hydra does not require libxml2
  support
* Add '--disable-silent-rules' to get more verbose output during
  the build

gsjaardema committed:
SEACAS: Add versions 2020-03-16 and 2020-01-16 (#15525)

eugeneswalker committed:
update checksum for intel-tbb@2020.1 (#15528)

jgalarowicz committed:
Previous qt package file changes broke the openspeedshop gui build.   (#15471)

* Previous qt changes broke the openspeedshop gui build.  This puts back the changes that caused the breakage.

* Update the qt version to be more robust.

Co-authored-by: Galarowicz, James <jgalarowicz@newmexicoconsortium.org>

ax3l committed:
openPMD-api: HDF5 1.12.0 Support (#15530)

Fix API breakage in HDF5 1.12.0 for released versions.

naromero77 committed:
QMCPACK Update March 2020  (#15511)

* Clean-up description of variants.

* Support AFQMC with CUDA.

bvanessen committed:
Added a package for the DiHydrogen distributed linear algebra library. (#15426)

* Added a package for the DiHydrogen distributed linear algebra library.

* Updated recipe to provide cuda architecture constaints.

* Addressed reviewer comments

* Fixed flake 8

ajkotobi committed:
Add package anaconda3 (#15404)

adamjstewart committed:
py-numpy: add 1.18.2 (#15546)

Commit:
r-openssl requires C99. (#15541)

Commit:
Remove trailing newline. (#15539)

Commit:
xts: add some dependencies. (#15537)

gauthier12 committed:
adding the fortran option for hdf5 dependency for comptatibility with petsc requirement (#15536)

Commit:
xsetpointer: fix wrong dependency type. (#15534)

Commit:
triangle: add X11 dependency. (#15533)

darmac committed:
update util-linux to version: 2.33:2.35.1 (#15529)

gsjaardema committed:
CGNS: Version update (#15523)

* CGNS: Version update

The CGNS library has several new versions.   There was a format-changing change in 3.4.0 which was removed for 3.4.1.  The change was then added again and released with a change to the major version (4.0.0).  Note that 4.0.0 should be close to the functionality of 3.4.0.

* CGNS: Add shared variant

Added the `shared`variant to make the CMake build correctly pick up the RPATH settings.

darmac committed:
add new package: userspace-rcu (#15515)

darmac committed:
add new package: acl (#15514)

btovar committed:
Cctools 7.1.0 (#15506)

* update to 7.1.0

* adds gettext as dependency (needed to link with python)

* disable python2 or python3 as needed

Commit:
revbayes: Update pre-build process (#15485)

Sinan81 committed:
Use python extend_path as pyqt sip fix (#15297)

* try extend path to solve PyQt5.sip not found issue

* disable private sip installation in sippackage class

* undo manual PyQt5 dir creation in py-sip site-packages dir

* fix typo

* fix typo

* also apply fix to PyQt4

* tidy up

* flake8 and tidy up

* tidy and undo hardcoding of python_include_dir

* replace hardcoded python inc dir

* fix minor issues

* rethink include dir variable name

* improve style

* add new versions

* implement new sip setup to qsci installation

* set sip-incdir correctly for the new setup

* setup extend_path thing before qsci python bindings

* take care of conflict

* flake8

* also extend for PyQt4

* improve style

* improve style

* SipPackage build sys should depend on py-sip

* consolidate extend_path fixes into SipPackage

* fix typo

* fix bugs

* flake8

* revert sip doc to pre-resource setup

* import os module

* flake8

Co-authored-by: Sinan81 <sbulut@3vgeomatics.com>

Sinan81 committed:
package/QGIS: fix cmake/FindPyqt.py (#15455)

* fix cmake/FindPyqt.py

* apply patch

* generalize patch

* rename patch

* fix typo

Co-authored-by: Sinan81 <sbulut@3vgeomatics.com>
Co-authored-by: Sinan81 <Sinan81@github>

gauthier12 committed:
adol-c:updating sources location (#15532)

* adol-c:updating sources location

* Update var/spack/repos/builtin/packages/adol-c/package.py

Co-Authored-By: Adam J. Stewart <ajstewart426@gmail.com>

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

bonachea committed:
upcxx: ensure we pickup the correct Python

ktsai7 committed:
modified spackage for flecsi and cinch (#15555)

adamjstewart committed:
py-horovod: set appropriate build env vars (#15548)

* py-horovod: set appropriate build env vars

* mxnet: add maintainer

* py-horovod: fetch git submodules

* py-torch: fix CUDA conflicts

balay committed:
petsc@3.7 fix dependency of superlu-dist@5.1.3 (#15551)

neilflood committed:
Tuiview movetogithub (#15498)

* py-tuiview: Source has moved to github

* py-tuiview: Explicitly require +python on gdal dependency

* py-tuiview: Versions up to 1.1.99 are qt4 only

* py-tuiview: Add version 1.2.6, which is qt5 only

* Explicit version range on gdal dependency

KineticTheory committed:
Trilinos: Add more variants (~netcdf ~matio ~glm) (#15483)

* Trilinos: Add more variants

+ Provide three new variants to allow building trilinos without netcdf, matio,
  or glm.
+ No change to defaults.

* Fix style issue.

germasch committed:
Upstreams: don't write metadata directory to upstream DB (#15526)

When trying to use an upstream Spack repository, as of f2aca86 Spack
was attempting to write to the upstream DB based on a new metadata
directory added in that commit. Upstream DBs are read-only, so this
should not occur.

This adds a check to prevent Spack from writing to the upstream DB

germasch committed:
Upstreams: don't write metadata directory to upstream DB (#15526)

When trying to use an upstream Spack repository, as of f2aca86 Spack
was attempting to write to the upstream DB based on a new metadata
directory added in that commit. Upstream DBs are read-only, so this
should not occur.

This adds a check to prevent Spack from writing to the upstream DB

ax3l committed:
Split: Fabtest/libfabric (#15557)

Fabtests provides runtime analysis tools and examples of libfabric.

As with other projects that are tightly version-bound, e.g.
`py-adios` and `adios`, the fact that releases stem from the same
repo does not imply they should be the same package.

Remove resources, which complicate the libfabric build, and update
the fabtests package accordingly.

mwkrentel committed:
hpctoolkit: restore binutils 2.34 for master (#15556)

Hpctoolkit master was recently updated to test for and allow old
binutils <= 2.33.1 and/or new binutils 2.34.  Older hpctoolkit up to
2030.03.01 will forever require :2.33.1.

Adjust the libunwind dependency for safety with the current
concretizer.

gartung committed:
Buildcache: fix relocation of binaries with relative rpaths (#15558)

adamjstewart committed:
py-matplotlib: add 3.2.1 (#15560)

bonachea committed:
upcxx: workaround a cray module problem observed on ALCF theta

Without this workaround, every invocation of the raw cc/CC compilers
within the Spack build environment results in a fatal error from
Cray's compiler driver, such as:

```
  cray-libsci/19.06.1 is not available for the INTEL compiler.

```
The cray-libsci module is completely irrelevant to UPC++ installation
anyhow, so tweak the environment to prevent Cray's compiler driver
from attempting to use it.

Commit:
amdblis + blis: fix spack info and libs query (#15566)

darmac committed:
update attr version to 2.4.48 (#15535)

* update attr version to 2.4.48

* attr: remove redundant make() in install()

michaelkuhn committed:
openssl: added v1.1.1e and v1.0.2u (#15568)

ystradmann committed:
Add package: py-deprecated (#15571)

https://github.com/tantale/deprecated

darmac committed:
add new package: yarn (#15562)

darmac committed:
add new package: ntirpc (#15561)

germasch committed:
petsc fix hdf5 (#15572)

* petsc: add checksum for 3.12.4

* petsc: constrain hdf5 to <= 1.10.x

Current petsc will error when being build with hdf5 1.12, so this ensures that
a compatible hdf5 will be used. Fix suggested by @balay.

j-ogas committed:
add charliecloud/0.14; remove umoci/skopeo; remove builder variant (#15573)

* add charliecloud/0.14; remove umoci/skopeo; remove builder variant

* fix typo

* add maintainer; add explicit config doc options; add py-requests as run dependency

* add autoconf build dependencies

teonnik committed:
cosma: added package at v2.0.2 (#15497)

* [new package] COSMA

* Use virtual dependencies

becker33 committed:
Cray bugfix: TERM missing while reading default target (#15381)

Bug: Spack hangs on some Cray machines

Reason: The TERM environment variable is necessary to run bash -lc "echo $CRAY_CPU_TARGET", but we run that command within env -i, which wipes the environment.

Fix: Manually forward the TERM environment variable to env -i /bin/bash -lc "echo $CRAY_CPU_TARGET"

becker33 committed:
Cray bugfix: TERM missing while reading default target (#15381)

Bug: Spack hangs on some Cray machines

Reason: The TERM environment variable is necessary to run bash -lc "echo $CRAY_CPU_TARGET", but we run that command within env -i, which wipes the environment.

Fix: Manually forward the TERM environment variable to env -i /bin/bash -lc "echo $CRAY_CPU_TARGET"

nithintsk committed:
Update MVAPICH2 package for 2.3.3 release (#15578)

Co-authored-by: senthilkumar.16 <senthilkumar.16@nowlab.cse.ohio-state.edu>

bonachea committed:
upcxx: support flags

darmac committed:
add new package: libiscsi (#15579)

darmac committed:
add new package: bcache (#15563)

* add new package: bcache

* add depends: gettext for libintl

Commit:
Improvements to Rust's Installation (#15395)

* Add sources and resources for each version of Rust

* install bootstrapping compiler into stage

* Add libgit2

* Install bootstrapping compiler correctly

* implement full rust bootstrap

* Remove support for Rust pre-1.14

Also add lots of comments

* Support only Rust 1.17 and newer

* Remove < 1.23 versions of Rust

* Change the layout of rust_releases for maintainability

* Remove LLVM variant

* Address flake8 issues

* Make libgit2 curl variant default False, conflict 0.28 and newer

* Remove binutils dependency

* Add ARM64 while we're at it

* flake8

* use the 'python' routine rather than relying on the correct python to be picked up

khuck committed:
perfstubs: added new package (#15547)

healther committed:
htop: add missing python build dependency (#15585)

HadrienG2 committed:
acts-core: added v0.18, updated boost dependency (#15584)

adamjstewart committed:
py-tfdlpack: add new package (#15583)

adamjstewart committed:
py-networkx: add 2.4 (#15582)

adamjstewart committed:
py-decorator: added v4.4.2 (#15581)

darmac committed:
add new package lksctp-tools (#15564)

Co-Authored-By: Massimiliano Culpo <massimiliano.culpo@gmail.com>

darmac committed:
add new package: glusterfs (#15565)

* add new package: glusterfs

* fix list_depth for url

* remove list_depth

* add list_url for glusterfs

mdorier committed:
Added two new versions of spdlog (#15586)

ChristianTackeGSI committed:
Show the packaga name for a missing patch sha256 (#15441)

When a patches sha256 is missing, also show the
packagename, so that analyzing is easier.

ChristianTackeGSI committed:
root: Add version 6.20.{00,02} (#15487)

* Add version 6.20.{00,02}, don't yet mark it preferred
* It needs zstd
* It needs numpy (at least for 6.20.00:6.20.03)
* Reorder python dependencies a bit

* Add mlp variant, default False
  Older versions always include mlp, so no conflicts there.
* Disable tmva, because it needs mlp
* tmva needs mlp, so add conflict

kmanalo committed:
papi: add version 6.0.0 (#15590)

bfovet committed:
Add NLopt version 2.6.1 (#15595)

adamjstewart committed:
py-dgl: add new package (#15594)

adamjstewart committed:
sentencepiece: add new package (#15596)

eugeneswalker committed:
tau: add dependency on hwloc (#15589)

tgamblin committed:
version bump: 0.14.1

tgamblin committed:
update CHANGELOG.md for 0.14.1

michaelkuhn committed:
spack checksum: Use package's fetch_options (#15481)

This makes sure that a package's fetch_options are used when fetching
new versions to checksum. This allows working around problems with
slow servers or those requiring a cookie to be set.

becker33 committed:
multiprocessing: allow Spack to run uninterrupted in background (#14682)

Spack currently cannot run as a background process uninterrupted because some of the logging functions used in the install method (especially to create the dynamic verbosity toggle with the v key) cause the OS to issue a SIGTTOU to Spack when it's backgrounded.

This PR puts the necessary gatekeeping in place so that Spack doesn't do anything that will cause a signal to stop the process when operating as a background process.

becker33 committed:
multiprocessing: allow Spack to run uninterrupted in background (#14682)

Spack currently cannot run as a background process uninterrupted because some of the logging functions used in the install method (especially to create the dynamic verbosity toggle with the v key) cause the OS to issue a SIGTTOU to Spack when it's backgrounded.

This PR puts the necessary gatekeeping in place so that Spack doesn't do anything that will cause a signal to stop the process when operating as a background process.

danlipsa committed:
Add version 5.8 and fix sqlite+column_metadata mismatch. (#15450)

adamjstewart committed:
py-sentencepiece: add new package (#15601)

carsonwoods committed:
Add 33.2.1 version with checksum to lua-luaposix (#15600)

bfovet committed:
Add Ceres Solver v1.14.0 with new variants (#15597)

* Add Ceres Solver version 1.14.0

* Add variants for Ceres Solver

bfovet committed:
Add gmsh v4.5.4 with new options (#15591)

* Add gmsh v4.5.4 with new options

This adds OpenCASCADE as an alternative to the oce package.
A new variant 'privateapi' is added to enable the gmsh private API.

* Make oce conflict with opencascade in gmsh

maiterth committed:
freeipmi: Initial commit (#15576)

* freeipmi: Initial commit

Added freeipmi package. Configure and build work fine, install only as root,
thus added warning and added reference to issue #4432 .

* freeipmi: fixed change requests

bonachea committed:
upcxx: Add git fetch of primary branches

adamjstewart committed:
py-torchtext: add new package (#15604)

nightlark committed:
swig: add version 4.0.1 (#15603)

nightlark committed:
fix: change typo dependnecies to dependencies (#15602)

gartung committed:
Buildcache: add unit tests for normalized path functions in relocate.py (#15607)

aweits committed:
performance: add a verification file to the database (#14693)

Reading the database repeatedly can be quite slow.  We need a way to speed
up Spack when it reads the DB multiple times, but the DB has not been
modified between reads (which is nearly all the time).

- [x] Add a file containing a unique uuid that is regenerated at database
    write time. Use this uuid to suppress re-parsing the database
    contents if we know a previous uuid and the uuid has not changed.

- [x] Fix mutable_database fixture so that it resets the last seen
    verifier when it resets.

- [x] Enable not rereading the database immediately after a write. Make
    the tests reset the last seen verifier in between tests that use the
    database fixture.

- [x] make presence of uuid module optional

bonachea committed:
upcxx: flake8 syntax fixup

PHHargrove commented on var/spack/repos/builtin/packages/upcxx/package.py:
```diff
@@ -43,12 +43,15 @@ class Upcxx(Package):
 
     conflicts('cross=none', when='platform=cray',
-              msg='None is unacceptable on Cray. Please specify an appropriate "cross" value')
+              msg='None is unacceptable on Cray.' +
```
Without quotes or context "None is unacceptable" seems easy to misread as its English meaning rather than a literal token.

tgamblin committed:
Merge branch 'releases/v0.14' into develop

darmac committed:
add new package: nfs-ganesha (#15580)

* add new pacakge: nfs-ganesha

* refine package.py

* update bison & flex as build depends

adamjstewart committed:
Fix bug in py-dgl installation, add unit tests (#15617)

adamjstewart committed:
bash: add patches up to 5.0.16 (#15615)

gartung committed:
Buildcache: attempt to build and install buildcaches in test environment (#15577)

* Put bindist.py on it's own branch

* Working on SL7 with python 3.6

* Flake8

* Put back clearing of compiler cache

drbenmorgan committed:
davix: add cxxstd variant (#15593)

* davix: add cxxstd variant

Davix is written in C++, so add this variant to allow dependents to specify
this so a consistent ABI is used.

* davix: fix flake8 errors
